src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -210,44 +240,59 @@ public bool ParseHeaders(TRequestHandler handler, ref SequenceReader<byte> reade
                         // the bounds check for the next lookup of span[length]
                         if ((uint)length < (uint)span.Length)
                         {
-                            // Early memory read to hide latency
-                            var expectedCR = span[length];
-                            // Correctly has a CR, move to next
-                            length++;
+                            var headerSpan = span[..length]; 
 
-                            if (expectedCR != ByteCR)
+                            if (length < 4)
                             {
-                                // Sequence needs to be CRLF not LF first.
-                                RejectRequestHeader(span[..length]);
+                                // Less than min possible headerSpan of 3 bytes a:b
```

Comment:
This logic throws for a header `"a:b\n"` even when `_enableLineFeedTerminator` is set. We should add a test case for this.